directory_header_footer/context_processors.py

```diff
@@ -1,17 +1,19 @@
+from functools import partial
+
 from django.conf import settings
 
 from directory_constants.constants import urls as default_urls
 
 
 def sso_processor(request):
     url = request.build_absolute_uri()
-    login_url = settings.SSO_PROXY_LOGIN_URL
+    add_next = partial('{0}?next={url}'.format, url=url)
```

Comment:
how does this differ from `'{0}?next={1}'.format(login_url, url),`?

Reply:
We need to check if the `next` is already specified in the `url` - then add only if not